app/routes.py

```diff
@@ -1,2 +1,234 @@
+    if task:
+        task.completed_at = None
+    # only send part that i am substituting 
+    # need to refractor to make patch all fit on one line
+        db.session.commit()
+        return jsonify({
+            "task":{
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.convert_complete()
+            }
+            }), 200
+    return "", 404    
+
+@goal_bp.route("", methods=["POST"], strict_slashes=False)
+def add_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return jsonify({
+        "details": "Invalid data"
+        }), 400
+
+    new_goal= Goal(title=request_body["title"])
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify({
+            "goal":{
+            "id": new_goal.goal_id,
+            "title": new_goal.title
+    }}), 201
+
+@goal_bp.route("", methods=["GET"], strict_slashes=False)
+def goal_index():
+    goals = Goal.query.all()
+    goal_response = []
+    for goal in goals:
+            goal_response.append(goal.to_json())
+
+    return jsonify(goal_response), 200
+
+@goal_bp.route("/<goal_id>", methods=["GET"], strict_slashes=False)
+def get_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal: 
+        return jsonify({
+            "goal":{
+            "id": goal.goal_id,
+            "title": goal.title
+            }
+            }), 200
+    return "", 404
+
+@goal_bp.route("/<goal_id>", methods=["PUT"], strict_slashes=False)
+def update_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        form_data = request.get_json()
+        goal.title = form_data["title"]
+        db.session.commit()
+        return jsonify({
+            "goal":{
+            "id": goal.goal_id,
+            "title": goal.title
+            }
+            }), 200
+    return "", 404
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"], strict_slashes=False)
+def delete_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        db.session.delete(goal)
+        db.session.commit()
+        return jsonify({
+            "details": (f'Goal {goal.goal_id} "{goal.title}" successfully deleted')
+        }), 200
+    return "", 404
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes=False)
+def assign_goal_to_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    request_body = request.get_json()
+    task_list = request_body["task_ids"]
+
+    for task_id in task_list:
+        task = Task.query.get(task_id)
+        if task:
+            task.goal_id = goal_id
+    
+    return jsonify({
+        "id": goal.goal_id,
+        "task_ids": task_list
+    }), 200
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"], strict_slashes=False)
+def get_tasks_for_one_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        #task_found = Task.query.filter_by(goal_id = goal_id)
```

Comment:
It's not a big deal but it's industry best practice to clean up commented out code before you create a PR.